packages/applications-service-api/src/controllers/advice.js

```diff
@@ -30,36 +29,22 @@ const getAdvice = async (req, res) => {
 };
 
 const getAdviceById = async (req, res) => {
+	logger.debug(`Retrieving advice by ID...`);
+
 	const { adviceID } = req.params;
-	try {
-		logger.debug(`Retrieving advice by ID...`);
+	const caseReference = req?.query?.caseReference;
 
-		const advice = await getAdviceByIdService(adviceID);
+	if (!caseReference) {
+		throw ApiError.badRequest('missing required parameter: caseReference');
```

Comment:
If the `caseReference` is marked as 'required' in the Open API spec then this check should already be handled at the router level (if the openapi validator middleware function is there) and you can safely destructure it from req.query